app/models/course.rb

```diff
@@ -466,4 +471,11 @@ def validate_applications_open_from
   def valid_date_range
     recruitment_cycle.application_start_date..recruitment_cycle.application_end_date
   end
+
+  def validate_program_type
+    if provider.present? && program_type_options.exclude?(program_type.to_sym)
```

Comment:
I think that the guard here, `provider.present?`, should actually be on the `validate` declaration up around line 155 of this file:

```
  validate :validate_program_type, on: :update, if: -> { |c| c.provider.present? }
```

All the more reason to have those `validate` declarations on separate lines.

Reply:
`provider.present?` That was a test quirk... in reality a course will always have a provider...

Reply:
was going to ask you about it at someday...

Reply:
Done.